test/libyul/yulControlFlowGraph/function.yul

```diff
@@ -28,10 +28,9 @@
 // Block0 [label="\
 // i: [ RET[i] ] => [ TMP[i, 0] TMP[i, 1] ]\l\
 // Assignment(x, y): [ TMP[i, 0] TMP[i, 1] ] => [ x y ]\l\
-// h: [ RET[h] x ] => [ ]\l\
-// h: [ RET[h] y ] => [ ]\l\
+// h: [ x ] => [ ]\l\
 // "];
-// Block0Exit [label="MainExit"];
+// Block0Exit [label="Terminated"];
```

Comment:
How well tested is Yul control flow graph? Do we have enough cases to be confident that terminating paths are detected correctly in corner cases? There's not much in `yulControlFlowGraph/`.

Reply:
It's exhaustively unit-tested, but generally still very well tested, since every semantics test is run through it...
Whether a user-defined function always reverts is just based on ``ControlFlowSideEffectsCollector``, though, and that has unit tests in ``test/libyul/controlFlowSideEffects/``.

Reply:
Semantic tests would not really detect the case where we wrongly think that the function can continue but it can't or vice-versa, right? It just affects this small optimization, which is optional. I'm just worried that it might not be working correctly in all corner cases and we're just not seeing it.